homeassistant/components/nanogrid_air/api.py

```diff
@@ -0,0 +1,64 @@
+"""Fetch API for Nanogrid Air."""
+
+import logging
+import socket
+
+import aiohttp
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def get_ip():
+    """Fetch IP address from hostname."""
+    try:
+        ip = socket.gethostbyname("ctek-ng-air.local")
+        _LOGGER.debug("Resolved IP: %s", ip)
+        return ip
+    except socket.gaierror as e:
+        _LOGGER.error("Failed to resolve hostname: %s", e)
+        return
+
+
+async def fetch_mac():
+    """Fetch mac address from status API."""
+    ip = await get_ip()
+
+    url_status = f"http://{ip}/status/"
```

Comment:
While it seems technically safe, checking the IP return (so that it is not null) before sending get request might be an idea? As it seems now, get_ip() can return blank in case of an exception and at this point there is no point sending out a request? :)

Reply:
Fixed